.github/workflows/tests.yaml

```diff
@@ -16,3 +29,88 @@ jobs:
         cache: 'maven'
     - name: Build with Maven
       run: cd stackrox-container-image-scanner && ./mvnw -B verify package hpi:hpi cyclonedx:makeAggregateBom
+    - uses: actions/upload-artifact@v3
+      with:
+        name: stackrox-container-image-scanner.hpi
+        path: stackrox-container-image-scanner/target/stackrox-container-image-scanner.hpi
+
+  e2e:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/checkout@v3
+      with:
+        repository: stackrox/stackrox
+        path: stackrox
+    - uses: docker/setup-buildx-action@v2
+    - uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        java-version: '11'
+        cache: 'gradle'
+    - name: Install kubectl
+      run: sudo snap install kubectl --classic
+    - name: Install gcloud
+      run: |
+        sudo snap install google-cloud-cli --classic
+        echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | sudo tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+        curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+        sudo apt-get update
+        sudo apt-get install google-cloud-sdk-gke-gcloud-auth-plugin
+    - uses: actions/download-artifact@v3
+      with:
+        name: stackrox-container-image-scanner.hpi
+        path: jenkins
+    - name: Build jenkins image
+      uses: docker/build-push-action@v4
+      with:
+        tags: jenkins-test
+        context: jenkins
+        push: false
+        load: true
+    - name: Run jenkins in background
+      run: docker run -d --add-host host.docker.internal:host-gateway -p 8080:8080 jenkins-test
+    - name: Create GKE cluster
+      id: create-cluster
+      env:
+        OPENSHIFT_CI: true
+        GCLOUD_SERVICE_ACCOUNT_OPENSHIFT_CI_ROX: ${{ secrets.GCLOUD_SERVICE_ACCOUNT_OPENSHIFT_CI_ROX }}
+      run: |
+        cd stackrox
+        export BUILD_ID=${GITHUB_RUN_ID}
+        export JOB_NAME=${GITHUB_JOB}
+        source "scripts/ci/gke.sh"
+        provision_gke_cluster "jenkins-plugin-e2e" 3 e2-standard-4
+        echo "CLUSTER_NAME=${CLUSTER_NAME}" >> $GITHUB_OUTPUT
+        wait_for_cluster
+    - name: Deploy Stackrox
+      id: deploy
+      run: |
+        cd stackrox
+        echo '#!/usr/bin/env bash\necho "Skipping sensor installation."' > deploy/k8s/sensor.sh
```

Comment:
Did you try `deploy/k8s/central.sh`? And also `LOAD_BALANCER=lb`? Then you could reuse some of the logic in `tests/e2e/lib.sh`. I'm particularly thinking about `wait_for_api()` to ensure central is ready.